packages/hardhat-core/src/internal/util/download.ts

```diff
@@ -9,6 +9,19 @@ interface FetchOptions {
   agent?: undefined | HttpsProxyAgent.HttpsProxyAgent;
 }
 
+const TEMP_FILE_PREFIX = "tmp-";
+
+function resolveTempFileName(filePath: string): string {
+  const { root, dir, ext, name } = path.parse(filePath);
+
+  return path.format({
+    root,
```

Comment:
Small comment I missed the previous time I reviewed this:

According to [the docs](https://nodejs.org/api/path.html#pathformatpathobject), «pathObject.root is ignored if pathObject.dir is provided», so I think we can remove `root` from here and from the result of `path.parse`.

Reply:
Fix applied